Helper/Invoice.php

```diff
@@ -267,14 +280,40 @@ public function handleCaptureWebhook(Order $order, Notification $notification):
     {
         $invoiceFactory = $this->adyenInvoiceFactory->create();
         $adyenInvoice = $this->adyenInvoiceResourceModel->getAdyenInvoiceByCaptureWebhook($order, $notification);
-
-        if (is_null($adyenInvoice)) {
-            throw new \Exception(sprintf(
-                'Unable to find adyen_invoice linked to original reference %s, psp reference %s, and order %s',
+        $isFullAmountCaptured = $this->adyenDataHelper->originalAmount(
+            $notification->getAmountValue(),
+            $notification->getAmountCurrency()
+            ) === $order->getBaseGrandTotal();
+
+        if (is_null($adyenInvoice) && $order->canInvoice()) {
+                if ($isFullAmountCaptured) {
+                   $adyenInvoice = $this->createInvoiceFromWebhook($order, $notification);
```

Comment:
I've just noticed that `$adyenInvoice` is an `array` instead of `AdyenInvoice`. In this case we should set `$adyenInvoiceObject`:
```suggestion
                   $adyenInvoiceObject = $this->createInvoiceFromWebhook($order, $notification);
```
Then later on line 320, use the invoice we've just created or load an existing one from DB:

```php
        $adyenInvoiceObject = $adyenInvoiceObject ?? $invoiceFactory->load($adyenInvoice[InvoiceInterface::ENTITY_ID], InvoiceInterface::ENTITY_ID);
```